File: rumahsehat_web/src/main/java/apap/tugaskelompok/rumahsehat/model/AppointmentModel.java
package apap.tugaskelompok.rumahsehat.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.GenericGenerator;
import org.springframework.format.annotation.DateTimeFormat;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import java.io.Serializable;
import java.time.LocalDateTime;

import com.fasterxml.jackson.annotation.JsonFormat;

@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@JsonIgnoreProperties(value={"tagihan"}, allowSetters = true)
@Table(name = "appointment")
public class AppointmentModel implements Serializable {

    @Id
    @Size(max = 255)
    @GenericGenerator(name = "appointment_id", strategy = "apap.tugaskelompok.rumahsehat.model.AppointmentIdGenerator")
    @GeneratedValue(generator = "appointment_id")
    @Column(name = "kode", nullable = false)
    private String kode;

    @NotNull
    @Column(name = "waktu_awal", nullable = false)
    @DateTimeFormat(pattern = "yyyy-MM-dd'T'HH:mm")
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm")
    private LocalDateTime waktuAwal;

    @NotNull
    @Column(name = "is_done", nullable = false)
    private Boolean isDone;

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "uuid_pasien", referencedColumnName = "uuid", nullable = false)
    private PasienModel pasien;

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "uuid_dokter", referencedColumnName = "uuid")
    private DokterModel dokter;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "kode_tagihan", referencedColumnName = "kode")
    private TagihanModel tagihan;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "id_resep", referencedColumnName = "id")
    @JsonIgnore
    private ResepModel resep;
}

File: rumah_sehat_mobile/appointment/lib/models/appointment_model.dart
class AppointmentModel {
  String kode;
  String waktuAwal;
  bool isDone;
  PasienModel pasien;
  DokterModel dokter;
  String? kodeTagihan;
  String? idResep;

  AppointmentModel(
      {required this.kode,
      required this.waktuAwal,
      required this.isDone,
      required this.pasien,
      required this.dokter,
      this.kodeTagihan,
      this.idResep});

  factory AppointmentModel.fromJson(Map<String, dynamic> json) =>
      AppointmentModel(
          kode: json['kode'],
          waktuAwal: json['waktuAwal'],
          isDone: json['isDone'],
          pasien: PasienModel.fromJson(json['pasien']),
          dokter: DokterModel.fromJson(json['dokter']),
          kodeTagihan: json['kode_tagihan'],
          idResep: json['id_resep']);
}

class DokterModel {
  String uuid;
  String nama;
  int tarif;

  DokterModel({required this.uuid, required this.nama, required this.tarif});

  factory DokterModel.fromJson(Map<String, dynamic> json) =>
      DokterModel(uuid: json['uuid'], nama: json['nama'], tarif: json['tarif']);
}

class PasienModel {
  String uuid;
  String nama;

  PasienModel({
    required this.uuid,
    required this.nama,
  });

  factory PasienModel.fromJson(Map<String, dynamic> json) =>
      PasienModel(uuid: json['uuid'], nama: json['nama']);
}
